File: goblin-api/src/main/java/org/goblinframework/api/dao/Id.java
package org.goblinframework.api.dao;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Id {

  Generator value();

  enum Generator {
    NONE,
    OBJECT_ID,
    AUTO_INC
  }
}

File: goblin-api/src/main/java/org/goblinframework/api/dao/Id.kt
package org.goblinframework.api.dao

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class Id(val value: Generator) {

  enum class Generator {
    NONE,
    OBJECT_ID,
    AUTO_INC
  }
}
